$OpenBSD: patch-library_rubygems_package_rb,v 1.1 2013/10/26 23:44:58 jeremy Exp $

Backport rubygems patch caa95666f698a4a5f1c73216f357e5b1b7216fb3 to fix
installation of .dotfiles in root of gems.

--- library/rubygems/package.rb.orig	Mon Oct 21 21:30:47 2013
+++ library/rubygems/package.rb	Mon Oct 21 21:31:32 2013
@@ -339,13 +339,9 @@ EOM
   def extract_tar_gz io, destination_dir, pattern = "*" # :nodoc:
     open_tar_gz io do |tar|
       tar.each do |entry|
-        # Some entries start with "./" which fnmatch does not like, see github
-        # issue #644
-        full_name = entry.full_name.sub %r%\A\./%, ''
+        next unless File.fnmatch pattern, entry.full_name, File::FNM_DOTMATCH
 
-        next unless File.fnmatch pattern, full_name
-
-        destination = install_location full_name, destination_dir
+        destination = install_location entry.full_name, destination_dir
 
         FileUtils.rm_rf destination
 
